flink-cdc-composer/src/test/java/org/apache/flink/cdc/composer/flink/FlinkPipelineTransformITCase.java

```diff
@@ -354,6 +355,93 @@ void testMultiTransformMissingProjection(ValuesDataSink.SinkApi sinkApi) throws
     }
 
+    @ParameterizedTest
+    @EnumSource
+    void testMultiTransformSchemaColumnsCompatibility(ValuesDataSink.SinkApi sinkApi) {
+        TransformDef nullProjection =
+                new TransformDef(
+                        "default_namespace.default_schema.mytable2",
+                        null,
+                        "age < 18",
+                        null,
+                        null,
+                        null,
+                        null);
+
+        TransformDef emptyProjection =
+                new TransformDef(
+                        "default_namespace.default_schema.mytable2",
+                        "",
+                        "age < 18",
+                        null,
+                        null,
+                        null,
+                        null);
+
+        TransformDef asteriskProjection =
+                new TransformDef(
+                        "default_namespace.default_schema.mytable2",
+                        "*",
+                        "age < 18",
+                        null,
+                        null,
+                        null,
+                        null);
+
+        List<TransformDef> transformDefList =
+                Arrays.asList(nullProjection, emptyProjection, asteriskProjection);
+
+        for (TransformDef transformDef : transformDefList) {
+            assertThatThrownBy(
+                            () ->
+                                    runGenericTransformTest(
+                                            sinkApi,
+                                            Arrays.asList(
+                                                    transformDef,
+                                                    new TransformDef(
+                                                            "default_namespace.default_schema.mytable2",
+                                                            // reference part column
+                                                            "id,UPPER(name) AS name",
+                                                            "age >= 18",
+                                                            null,
+                                                            null,
+                                                            null,
+                                                            null)),
+                                            Collections.emptyList()))
+                    .rootCause()
+                    .hasMessage(
+                            "Unable to merge schema columns={`id` BIGINT,`name` VARCHAR(255),`age` TINYINT,`description` STRING}, primaryKeys=id, options=() "
+                                    + "and columns={`id` BIGINT,`name` STRING}, primaryKeys=id, options=() with different column counts.");
+        }
+
+        for (TransformDef transformDef : transformDefList) {
+            assertThatThrownBy(
+                    () ->
```

Comment:
Also add some chaining assertions to make sure we're catching correct Exception and messages?